Tests/TensorFlowTests/OptimizerTests.swift

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+@testable import TensorFlow
+
+final class OptimizerTests: XCTestCase {
+    func testRAdam() {
+        // The expected value of the gradient was computed using the following Python code
+        // Requirements: Tensorflow >= 2.1.0 and tf-addons (see https://github.com/tensorflow/addons)
+        // ```
+        // import tensorflow as tf
+        // from tensorflow_addons.optimizers import RectifiedAdam
+        // var_0 = tf.Variable([1.0, 2.0], dtype=tf.dtypes.float32)
+        // grad_0 = tf.Variable([0.1, 0.2], dtype=tf.dtypes.float32)
+        // grads_and_vars = list(zip([grad_0], [var_0]))
+        // optimizer = RectifiedAdam(lr=1e-3, epsilon=1e-8)
+        // for _ in range(1000):
+        //   optimizer.apply_gradients(grads_and_vars)
+        // print(var_0.read_value())
+        // >>> [0.5553605, 1.5548599]
+        // Current implementation = [0.5543607, [1.55286]]
+        // Difference of [0.0009997, 0.0019999].
+        // ```
```

Comment:
> @dan-zheng I've added the Python code in the test comments and shifted to the to-level `gradient` as suggested.

Thanks! I verified that the Python code produces the results in the comments (`[0.5553605, 1.5548599]`). 

I wonder if you have ideas about why there's a `[0.0009997, 0.0019999]` difference from Swift results?

What's the difference before your patch to `Sources/TensorFlow/Optimizers/MomentumBased.swift`? If the difference is lower with this patch, then that's empirical proof that this patch is in the right direction.

Reply:
I'm really not sure about the difference. I was thinking that there won't be any error since the original test in `addons` repo was using `eps=1e-7` as their default parameter, but still an error of `~1e-3` with `eps=1e-8` is something to look for. Regarding swift-format, all the CLI arguments aren't recognized. I'll look into it right now with a fresh build.

Reply:
@vballoli 
It seems there's another bug.

![](https://user-images.githubusercontent.com/18056781/69412657-58a72c00-0d35-11ea-9537-a689beae6500.png)
You are ignoring $\hat{m}_t$.

In `tensorflow_addons.optimizers.RectifiedAdam` it appears as `m_corr_t`.
https://github.com/tensorflow/addons/blob/v0.7.1/tensorflow_addons/optimizers/rectified_adam.py#L179

Reply:
I noticed there are several differences between TensorFlow implementation and PyTorch implementation.
https://github.com/tensorflow/addons/blob/v0.7.1/tensorflow_addons/optimizers/rectified_adam.py#L146-L209
https://github.com/LiyuanLucasLiu/RAdam/blob/master/radam/radam.py#L28-L94
PyTorch implementation is official, but it seems TensorFlow version is following the paper more strictly.

Current implementation is based on PyTorch. No wonder the test result doesn't match.

Reply:
My comments above were wrong.
I thought $\hat{m}_t$ doesn't appear in PyTorch version, but it is there in different form.
https://github.com/LiyuanLucasLiu/RAdam/blob/master/radam/radam.py#L159-L161

Reply:
Hi @t-ae, yeah the `* step` wasn't part of the algorithm, my bad. Sorry for the inactivity on this PR, I've not been able to get swift-format working ? Are there any alternatives ?

Reply:
> Sorry for the inactivity on this PR, I've not been able to get swift-format working ? Are there any alternatives ?

I resolved merge conflicts for you!

Sorry that `swift-format` isn't building for you - the team is discussing whether to bundle `swift-format` in Swift for TensorFlow toolchains for easier access.